examples/demo/01-snake/snake.c

```diff
@@ -309,7 +309,7 @@ SDL_AppResult SDL_AppInit(void **appstate, int argc, char *argv[])
         return SDL_APP_FAILURE;
     }
 
-    AppState *as = SDL_calloc(1, sizeof(AppState));
+    AppState *as = (AppState*)SDL_calloc(1, sizeof(AppState));
```

Comment:
Building this as C shouldn't error out and need not have this cast: why is it failing for you?

Reply:
I think its my failure, i built it with cpp.

Not have problem in C.

Reply:
Recently, sam applied changes to make code compile in C++ mode.
If the example can be built in C and C++ mode, then it's an even better example.
```suggestion
    AppState *as = (AppState *)SDL_calloc(1, sizeof(AppState));
```